sql/core/src/test/resources/sql-tests/results/interval.sql.out

```diff
@@ -0,0 +1,375 @@
+-- !query 18 schema
+struct<interval 4 weeks 2 days:calendarinterval>
+-- !query 18 output
+interval 4 weeks 2 days
+
+
+-- !query 19
+select 30 day day
+-- !query 19 schema
+struct<day:calendarinterval>
+-- !query 19 output
+interval 4 weeks 2 days
+
+
+-- !query 20
+select 30 day day day
+-- !query 20 schema
+struct<>
+-- !query 20 output
+org.apache.spark.sql.catalyst.parser.ParseException
+
+extraneous input 'day' expecting <EOF>(line 1, pos 18)
+
+== SQL ==
+select 30 day day day
+------------------^^^
+
+
+-- !query 21
+select date '2012-01-01' - 30 day
+-- !query 21 schema
+struct<CAST(CAST(DATE '2012-01-01' AS TIMESTAMP) - interval 4 weeks 2 days AS DATE):date>
+-- !query 21 output
+2011-12-02
+
+
+-- !query 22
+select date '2012-01-01' - 30 day day
+-- !query 22 schema
+struct<day:date>
+-- !query 22 output
+2011-12-02
+
+
+-- !query 23
+select date '2012-01-01' - 30 day day day
+-- !query 23 schema
+struct<>
+-- !query 23 output
+org.apache.spark.sql.catalyst.parser.ParseException
+
+extraneous input 'day' expecting <EOF>(line 1, pos 38)
+
+== SQL ==
+select date '2012-01-01' - 30 day day day
+--------------------------------------^^^
+
+
+-- !query 24
+select date '2012-01-01' + '-30' day
+-- !query 24 schema
+struct<CAST(CAST(DATE '2012-01-01' AS TIMESTAMP) + interval -4 weeks -2 days AS DATE):date>
+-- !query 24 output
+2011-12-02
+
+
+-- !query 25
+select date '2012-01-01' + interval '-30' day
+-- !query 25 schema
+struct<CAST(CAST(DATE '2012-01-01' AS TIMESTAMP) + interval -4 weeks -2 days AS DATE):date>
+-- !query 25 output
+2011-12-02
+
+
+-- !query 26
+select date '2012-01-01' + interval (-30) day
+-- !query 26 schema
+struct<>
+-- !query 26 output
+org.apache.spark.sql.AnalysisException
+Undefined function: 'interval'. This function is neither a registered temporary function nor a permanent function registered in the database 'default'.; line 1 pos 27
+
+
+-- !query 27
+select date '2012-01-01' + (-30) day
+-- !query 27 schema
+struct<>
+-- !query 27 output
+org.apache.spark.sql.AnalysisException
+cannot resolve '(DATE '2012-01-01' + -30)' due to data type mismatch: differing types in '(DATE '2012-01-01' + -30)' (date and int).; line 1 pos 7
+
+
+-- !query 28
+create temporary view t as select * from values (1), (2) as t(a)
+-- !query 28 schema
+struct<>
+-- !query 28 output
+
+
+
+-- !query 29
+select date '2012-01-01' + interval (a + 1) day from t
+-- !query 29 schema
+struct<>
+-- !query 29 output
+org.apache.spark.sql.AnalysisException
+Undefined function: 'interval'. This function is neither a registered temporary function nor a permanent function registered in the database 'default'.; line 1 pos 27
+
+
+-- !query 30
+select date '2012-01-01' + (a + 1) day from t
+-- !query 30 schema
+struct<>
+-- !query 30 output
+org.apache.spark.sql.AnalysisException
+cannot resolve '(DATE '2012-01-01' + (t.`a` + 1))' due to data type mismatch: differing types in '(DATE '2012-01-01' + (t.`a` + 1))' (date and int).; line 1 pos 7
```

Comment:
How about the columns having the matched data type?

Reply:
ok, I'll add tests to check